### ISSUE TITLE
Upgrading to haproxy 2.x

### ISSUE THREAD

**Issue by SEJeff** [jcmoraisjr/haproxy-ingress#367]
It would be nice to use haproxy 2.x and some of the features it provides in kubernetes.

**Comment by SEJeff** [comment]
The [haproxy:2.0.4-alpine](https://github.com/docker-library/haproxy/blob/2e948aa9880c8184c8e001bc5416cabf10d507f1/2.0/alpine/Dockerfile) image is available.

**Comment by jdconti** [comment]
Huge +1 for haproxy  2.x.

[Note: since HAProxy 2.0.0, the official source includes a Prometheus exporter module that can be built into your binary with a single flag during build time and offers an exporter-free Prometheus endpoint.](https://github.com/prometheus/haproxy_exporter)

**Comment by vijaykumark** [comment]
+1
It would be really useful if the ingress controller is upgraded to 2.x

**Comment by jcmoraisjr** [comment]
Hi there, thanks for using or evaluating HAProxy Ingress! I need at least one stable version per haproxy version and the schedule is as follow:

* v0.8 is haproxy 1.8.x and we'll start beta shortly. Really. The current code is quite stable and there is only a few missing pieces to implement. Follow #274 !
* v0.9 will be haproxy 1.9.x, will start just after releasing the first v0.8 beta, which should happen in a few days or weeks. This is a really fast release, I'll just implement some 1.9.x stuff like htx and h2/grpc backend
* v1.0 or v0.10 - I don't know yet - will be haproxy 2.0.x, implemented on top of the latest v0.9 code with htx, h2/grpc, prometheus and so on.

All these releases will happen almost in parallel. It's a bit hard to me give a release date because I'm not full time in haproxy ingress, but provided that v0.8 is almost beta and add htx/h2/grpc shouldn't be a hard job, I think we can see the first haproxy ingress beta with haproxy 2.0.x within a month or so. This should happen before my vacation that starts on oct/05.

**Comment by abh** [comment]
@jcmoraisjr This is really great, thank you! Why not skip haproxy v1.9 ?

**Comment by jcmoraisjr** [comment]
Hi @abh I chose that way because this is almost a zero-effort option, this won't postpone haproxy 2.0 usage for more than a few days, and I want to give option to one that, for any reason, have any problem with 2.0 and want to check with 1.9.

**Comment by jcmoraisjr** [comment]
Hi, v0.10 is now haproxy 2.0, focused mostly on metrics. [This example](http://haproxy-ingress.github.io/docs/examples/metrics/) has most of the changes in this version. Closing.
